contracts/ERC4626/VenusERC4626.sol

```diff
@@ -0,0 +1,260 @@
+        _checkVaultActionPaused(VaultAction.WITHDRAW);
+        if (assets > maxWithdraw(withdrawer)) {
+            revert ERC4626__WithdrawMoreThanMax();
+        }
+
+        uint256 shares = previewWithdraw(assets);
+        beforeWithdraw(assets, shares);
+        _withdraw(_msgSender(), receiver, withdrawer, assets, shares);
+        return shares;
+    }
+
+    /// @inheritdoc ERC4626
+    function redeem(uint256 shares, address receiver, address redeemer) public override returns (uint256) {
+        _checkVaultActionPaused(VaultAction.REDEEM);
+        if (shares > maxRedeem(redeemer)) {
+            revert ERC4626__RedeemMoreThanMax();
+        }
+
+        uint256 assets = previewRedeem(shares);
+        beforeWithdraw(assets, shares);
+        _withdraw(_msgSender(), receiver, redeemer, assets, shares);
+        return assets;
+    }
+
+    /// @notice Returns the maximum deposit allowed based on Venus supply caps.
+    /// @dev If minting is paused or the supply cap is reached, returns 0.
+    /// @param /*account*/ The address of the account.
+    /// @return The maximum amount of assets that can be deposited.
+    function maxDeposit(address /*account*/) public view virtual override returns (uint256) {
+        if (COMPTROLLER.actionPaused(address(VTOKEN), Action.MINT)) {
+            return 0;
+        }
+
+        uint256 supplyCap = COMPTROLLER.supplyCaps(address(VTOKEN));
+        uint256 totalSupply_ = VTOKEN.totalSupply();
+        return supplyCap > totalSupply_ ? supplyCap - totalSupply_ : 0;
+    }
+
+    /// @notice Returns the maximum amount of shares that can be minted.
+    /// @dev This is derived from the maximum deposit amount converted to shares.
+    /// @param /*account*/ The address of the account.
+    /// @return The maximum number of shares that can be minted.
+    function maxMint(address /*account*/) public view override returns (uint256) {
+        return convertToShares(maxDeposit(address(0)));
+    }
+
+    /// @notice Returns the maximum amount that can be withdrawn.
+    /// @dev The withdrawable amount is limited by the available cash in the vault.
+    /// @param receiver The address of the account withdrawing.
+    /// @return The maximum amount of assets that can be withdrawn.
+    function maxWithdraw(address receiver) public view virtual override returns (uint256) {
+        uint256 cash = VTOKEN.getCash();
+        uint256 assetsBalance = convertToAssets(balanceOf(receiver));
+        return cash < assetsBalance ? cash : assetsBalance;
+    }
+
+    /// @notice Returns the maximum amount of shares that can be redeemed.
+    /// @dev Redemption is limited by the available cash in the vault.
+    /// @param receiver The address of the account redeeming.
+    /// @return The maximum number of shares that can be redeemed.
+    function maxRedeem(address receiver) public view override returns (uint256) {
+        uint256 cash = VTOKEN.getCash();
```

Comment:
Shouldn't we check if the REDEEM action is paused in the vToken (similar to the check we are performing in the deposit flow)?